pkg/migration/runner.go

```diff
@@ -938,3 +990,53 @@ func (r *Runner) dumpStatus(ctx context.Context) {
 		}
 	}
 }
+
+func (r *Runner) sentinelTableExists(ctx context.Context) (bool, error) {
+	if r.sentinelTable == nil {
+		r.registerSentinelTable()
+	}
+	sql := "SELECT COUNT(*) FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?"
+	var sentinelTableExists int
+	err := r.db.QueryRowContext(ctx, sql, r.sentinelTable.SchemaName, r.sentinelTable.TableName).Scan(&sentinelTableExists)
+	if err != nil {
+		return false, err
+	}
+	return sentinelTableExists > 0, nil
+}
+
+// Check every sentinelCheckInterval up to sentinelWaitLimit to see if sentinelTable has been dropped
+func (r *Runner) handleDeferredCutOver(ctx context.Context) error {
+	if sentinelExists, err := r.sentinelTableExists(ctx); err != nil {
+		return err
+	} else if !sentinelExists {
+		// Sentinel table does not exist, we can proceed with cutover
+		return nil
+	}
+
+	r.logger.Warnf("cutover deferred while sentinel table %s exists; will wait %s", r.sentinelTable.QuotedName, sentinelWaitLimit)
+
+	timer := time.NewTimer(sentinelWaitLimit)
+
+	ticker := time.NewTicker(sentinelCheckInterval)
+	defer ticker.Stop()
+	logticker := time.NewTicker(10 * sentinelCheckInterval)
+	defer logticker.Stop()
```

Comment:
As well as renaming this func (wait on sentinel ..) we should use the global logger here. Which means adding a new status:

	r.setCurrentState(stateWaitingOnSentinelTable)


And the logging should be performed by `r.dumpStatus()`. This might require a little bit of work since you'll want the timer to be part of the struct so you can state the remaining time.

i.e. in dumpStatus it should have something like this:

```
			case stateWaitingOnSentinelTable:
				// We are waiting for the sentinel table to be dropped.
				// Give some really useful status info on it.
				r.logger.Infof("migration status: state=%s binlog-deltas=%v total-time=%s time-waiting=XX maximum-wait-time=XX conns-in-use=%d",
					r.getCurrentState().String(),
					r.replClient.GetDeltaLen(),
					time.Since(r.startTime).Round(time.Second),
					r.db.Stats().InUse,
				)
```